commit 2d61391270a3ceb95b3dd536ea13002e653323b6
Author: Sean Christopherson <seanjc@google.com>
Date:   Mon May 2 00:07:33 2022 +0200

    KVM: x86: Differentiate Soft vs. Hard IRQs vs. reinjected in tracepoint
    
    In the IRQ injection tracepoint, differentiate between Hard IRQs and Soft
    "IRQs", i.e. interrupts that are reinjected after incomplete delivery of
    a software interrupt from an INTn instruction.  Tag reinjected interrupts
    as such, even though the information is usually redundant since soft
    interrupts are only ever reinjected by KVM.  Though rare in practice, a
    hard IRQ can be reinjected.
    
    Signed-off-by: Sean Christopherson <seanjc@google.com>
    [MSS: change "kvm_inj_virq" event "reinjected" field type to bool]
    Signed-off-by: Maciej S. Szmigiero <maciej.szmigiero@oracle.com>
    Message-Id: <9664d49b3bd21e227caa501cff77b0569bebffe2.1651440202.git.maciej.szmigiero@oracle.com>
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 959d66b9be94..8109805b5429 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1405,7 +1405,7 @@ struct kvm_x86_ops {
 	u32 (*get_interrupt_shadow)(struct kvm_vcpu *vcpu);
 	void (*patch_hypercall)(struct kvm_vcpu *vcpu,
 				unsigned char *hypercall_addr);
-	void (*inject_irq)(struct kvm_vcpu *vcpu);
+	void (*inject_irq)(struct kvm_vcpu *vcpu, bool reinjected);
 	void (*inject_nmi)(struct kvm_vcpu *vcpu);
 	void (*queue_exception)(struct kvm_vcpu *vcpu);
 	void (*cancel_injection)(struct kvm_vcpu *vcpu);